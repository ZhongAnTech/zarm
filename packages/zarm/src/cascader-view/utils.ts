import { FieldNames } from '../picker-view/interface';
import { resolvedFieldNames } from '../picker-view/utils';
import type {
  BaseCascaderViewProps,
  CascaderItem,
  CascaderOption,
  CascaderValue,
} from './interface';

export const parseState = (props: BaseCascaderViewProps) => {
  const { value, defaultValue } = props;
  const currentValue: CascaderValue[] = [];

  return {
    value: value || defaultValue,
    currentValue,
  };
};

export const parseItems = (
  options: CascaderOption[],
  value: CascaderValue[],
  fieldNames?: Partial<FieldNames>,
) => {
  const field = resolvedFieldNames(fieldNames);
  const items: CascaderItem[] = [];
  const traverse = (memo: CascaderOption[], depth = 0) => {
    memo.forEach((option) => {
      const children = option[field.children] as CascaderOption[] | undefined;
      const currentValue = value[depth];
      if (option[field.value] === currentValue) {
        items[depth] = option;

        if (Array.isArray(children) && children.length > 0) {
          traverse(children, depth + 1);
        }
      }
    });
  };

  traverse(options);

  return items;
};
