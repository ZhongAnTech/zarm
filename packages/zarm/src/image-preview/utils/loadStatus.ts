enum LOAD_STATUS {
  before = 'loadBefore',
  start = 'loadStart',
  end = 'loadEnd',
  after = 'loadAfter',
}

export default LOAD_STATUS;
