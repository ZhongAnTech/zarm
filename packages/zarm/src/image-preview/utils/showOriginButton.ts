import type { Images } from '../PropsType';

const showOriginButton = (images: Images, index?: number): boolean => {
  if (images && typeof index === 'number' && images[index] && images[index].originUrl) {
    return true;
  }
  return false;
};

export default showOriginButton;
