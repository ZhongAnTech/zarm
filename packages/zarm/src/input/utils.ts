export function getValue(props, defaultValue: string | number = ''): string | number {
  if (typeof props.value !== 'undefined' && props.value !== null) {
    return props.value;
  }
  if (typeof props.defaultValue !== 'undefined' && props.defaultValue !== null) {
    return props.defaultValue;
  }
  return defaultValue;
}
