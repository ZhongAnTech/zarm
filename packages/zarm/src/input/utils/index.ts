export const regexAstralSymbols = /[\uD800-\uDBFF][\uDC00-\uDFFF]|\n/g;

export const countSymbols = (text = '') => {
  return text.replace(regexAstralSymbols, '_').length;
};
