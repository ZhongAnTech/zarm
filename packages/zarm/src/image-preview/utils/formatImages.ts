import { isObject, isString } from '../../utils/validate';
import type { Images, ImageObject } from '../PropsType';
import LOAD_STATUS from './loadStatus';

function isImageString(image: ImageObject | string): image is string {
  return isString(image);
}
function isImageObject(image: ImageObject | string): image is ImageObject {
  return isObject(image);
}

const formatImages = (images: ReadonlyArray<ImageObject | string>): Images => {
  const previewImages: Images = [];
  (images || []).forEach((image: ImageObject | string) => {
    if (isImageString(image)) {
      previewImages.push({
        url: image,
      });
    } else if (isImageObject(image)) {
      previewImages.push({
        url: image.url,
        originUrl: image.originUrl,
        loaded: LOAD_STATUS.before,
      });
    }
  });
  return previewImages;
};

export default formatImages;
