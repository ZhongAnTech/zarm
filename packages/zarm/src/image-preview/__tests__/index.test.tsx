/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable dot-notation */
import React from 'react';
import { mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImagePreview from '../ImagePreview';
import { images, originImages } from '../../../tests/testData/images';
// import Carousel from '../../carousel';
// import { Images } from '../interface';
import { sleep } from '../../../tests/utils';
// import ActivityIndicator from '../../activity-indicator';
// import ImagePreviewEnhanced from '../index';

describe('ImagePreview', () => {
  afterEach(() => {
    jest.restoreAllMocks();
    jest.useRealTimers();
  });
  describe('snapshot', () => {
    it('renders correctly', () => {
      const wrapper = render(<ImagePreview visible onChange={jest.fn()} images={images} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders correctly with origin', () => {
      const wrapper = render(<ImagePreview visible onChange={jest.fn()} images={originImages} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  // it('should pass locale and localeCode props to original component after wrapped by ConfigReceiverWrapper HOC', () => {
  //   /*  @ts-ignore */
  //   const wrapper = mount(<ImagePreviewEnhanced visible images={images} />);
  //   expect(wrapper.find(ImagePreview).prop('locale')).toEqual({
  //     loadBefore: '查看原图',
  //     loadStart: '加载中',
  //     loadEnd: '加载完成',
  //   });
  //   expect(wrapper.find(ImagePreview).prop('localeCode')).toEqual('zh-CN');
  // });

  // it('should get initial derived state from props - 2', () => {
  //   const wrapper = mount(<ImagePreview visible images={images} />);
  //   expect(wrapper.state()).toEqual({
  //     visible: true,
  //     activeIndex: 0,
  //     currentIndex: 0,
  //     images: [
  //       { url: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png' },
  //       { url: 'https://static.zhongan.com/website/health/zarm/images/banners/2.png' },
  //       { url: 'https://static.zhongan.com/website/health/zarm/images/banners/3.png' },
  //     ],
  //     prevVisible: true,
  //     prevActiveIndex: 0,
  //     prevImages: [
  //       'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //       'https://static.zhongan.com/website/health/zarm/images/banners/2.png',
  //       'https://static.zhongan.com/website/health/zarm/images/banners/3.png',
  //     ],
  //   });
  // });

  // it('should handle onChange event', async () => {
  //   const mOnChange = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} onChange={mOnChange} />);
  //   expect(wrapper.state('currentIndex')).toEqual(0);
  //   wrapper.find(Carousel).invoke('onChange')!(1);
  //   expect(wrapper.state('currentIndex')).toEqual(1);
  //   expect(mOnChange).toBeCalledWith(1);
  // });

  // it("should not call onChange event if it's not existed", async () => {
  //   const wrapper = mount(<ImagePreview visible images={images} />);
  //   expect(wrapper.state('currentIndex')).toEqual(0);
  //   wrapper.find(Carousel).invoke('onChange')!(1);
  //   expect(wrapper.state('currentIndex')).toEqual(1);
  // });

  // it('should load origin url for first image', () => {
  //   let onloadRef: typeof Image.prototype.onload;
  //   let srcRef: string;
  //   const onloadSymbol = Symbol('onload');
  //   const srcSymbol = Symbol('src');
  //   Object.defineProperty(Image.prototype, 'onload', {
  //     get() {
  //       return this[onloadSymbol];
  //     },
  //     set(onload) {
  //       onloadRef = onload;
  //       this[onloadSymbol] = onload;
  //     },
  //     configurable: true,
  //   });
  //   Object.defineProperty(Image.prototype, 'src', {
  //     get() {
  //       return this[srcSymbol];
  //     },
  //     set(src) {
  //       srcRef = src;
  //       this[srcSymbol] = src;
  //     },
  //     configurable: true,
  //   });
  //   jest.useFakeTimers();
  //   const wrapper = mount(<ImagePreview images={originImages} visible />);
  //   wrapper.find('.za-image-preview__origin__button').simulate('click');
  //   expect((wrapper.state('images') as Images)[0]).toEqual({
  //     url: 'https://cdn-health.zhongan.com/zarm/imagePreview/compress_1.png',
  //     originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //     loaded: 'loadStart',
  //   });
  //   expect(wrapper.find(ActivityIndicator).exists()).toBeTruthy();
  //   onloadRef();
  //   expect((wrapper.state('images') as Images)[0]).toEqual({
  //     url: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //     originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //     loaded: 'loadEnd',
  //   });
  //   jest.advanceTimersByTime(1500);
  //   expect((wrapper.state('images') as Images)[0]).toEqual({
  //     url: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //     originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  //     loaded: 'loadAfter',
  //   });
  //   expect(srcRef!).toEqual('https://static.zhongan.com/website/health/zarm/images/banners/1.png');
  // });

  it('should render show origin url button if origin url is not existed', () => {
    const wrapper = mount(<ImagePreview images={images} visible />);
    const buttonWrapper = wrapper.find('.za-image-preview__origin__button');
    expect(buttonWrapper.exists()).toBeFalsy();
  });

  // it("should not call onClose handler if user's touch is moving", () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
  //   expect(wrapper.instance()['moving']).toBeUndefined();
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('touchstart').simulate('touchmove').simulate('click');
  //   expect(wrapper.instance()['moving']).toBeTruthy();
  //   expect(mOnClose).not.toBeCalled();
  // });

  // it("should not call onClose handler if user's mouse is moving", () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
  //   expect(wrapper.instance()['moving']).toBeUndefined();
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('mousedown').simulate('mousemove').simulate('click');
  //   expect(wrapper.instance()['moving']).toBeTruthy();
  //   expect(mOnClose).not.toBeCalled();
  // });

  it("should call onClose handler if user's touch is NOT moving", () => {
    const mOnClose = jest.fn();
    const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
    const contentWrapper = wrapper.find('.za-image-preview__content');
    contentWrapper.simulate('click');
    expect(mOnClose).toBeCalled();
  });

  it('should not call onClose handler when touch end and the duration between touchstart and touchend greater than 300ms', async () => {
    const mOnClose = jest.fn();
    const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
    const contentWrapper = wrapper.find('.za-image-preview__content');
    contentWrapper.simulate('touchstart').simulate('touchmove');
    await sleep(500);
    contentWrapper.simulate('touchend');
    expect(mOnClose).not.toBeCalled();
  });

  // it('should call onClose handler when touch end and the duration between touchstart and touchend less than 300ms', async () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('touchstart');
  //   await sleep(200);
  //   jest.useFakeTimers();
  //   contentWrapper.simulate('touchend');
  //   jest.advanceTimersByTime(300);
  //   expect(mOnClose).toBeCalledTimes(1);
  //   expect(setTimeout).toBeCalledWith(expect.any(Function), 300);
  // });

  // it("should clear setTimeout schedule if user's touch is moving and previous scheduler exists", async () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} />);
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('touchstart');
  //   await sleep(200);
  //   jest.useFakeTimers();
  //   contentWrapper.simulate('touchend');
  //   // eslint-disable-next-line prefer-destructuring
  //   const doubleClickTimer = wrapper.instance()['doubleClickTimer'];
  //   contentWrapper.simulate('touchend');
  //   expect(clearTimeout).toBeCalledWith(doubleClickTimer);
  //   expect(mOnClose).not.toBeCalled();
  // });

  it('should not call onClose handler when touch end and the duration between touchstart and touchcancel greater than 300ms', async () => {
    const mOnClose = jest.fn();
    const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
    const contentWrapper = wrapper.find('.za-image-preview__content');
    contentWrapper.simulate('touchstart').simulate('touchmove');
    await sleep(500);
    contentWrapper.simulate('touchcancel');
    expect(mOnClose).not.toBeCalled();
  });

  // it('should call onClose handler when touch end and the duration between touchstart and touchcancel less than 300ms', async () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} onClose={mOnClose} />);
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('touchstart');
  //   await sleep(200);
  //   jest.useFakeTimers();
  //   contentWrapper.simulate('touchcancel');
  //   jest.advanceTimersByTime(300);
  //   expect(mOnClose).toBeCalledTimes(1);
  //   expect(setTimeout).toBeCalledWith(expect.any(Function), 300);
  // });

  // it("should clear setTimeout schedule if user's touch is moving and previous scheduler exists", async () => {
  //   const mOnClose = jest.fn();
  //   const wrapper = mount(<ImagePreview visible images={images} />);
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('touchstart');
  //   await sleep(200);
  //   jest.useFakeTimers();
  //   contentWrapper.simulate('touchcancel');
  //   // eslint-disable-next-line prefer-destructuring
  //   const doubleClickTimer = wrapper.instance()['doubleClickTimer'];
  //   contentWrapper.simulate('touchcancel');
  //   expect(clearTimeout).toBeCalledWith(doubleClickTimer);
  //   expect(mOnClose).not.toBeCalled();
  // });

  // it('should set moving to false and touchStartTime to 0 when mouse up', () => {
  //   const wrapper = mount(<ImagePreview visible images={images} />);
  //   const contentWrapper = wrapper.find('.za-image-preview__content');
  //   contentWrapper.simulate('mousedown').simulate('mousemove');
  //   expect(wrapper.instance()['moving']).toBeTruthy();
  //   expect(wrapper.instance()['touchStartTime']).toBeGreaterThan(0);
  //   jest.useFakeTimers();
  //   contentWrapper.simulate('mouseup');
  //   jest.advanceTimersByTime(1);
  //   expect(wrapper.instance()['moving']).toBeFalsy();
  //   expect(wrapper.instance()['touchStartTime']).toBe(0);
  // });

  it('should render pagination', () => {
    const wrapper = mount(<ImagePreview visible images={images} />);
    const paginationWrapper = wrapper.find('.za-image-preview__index');
    expect(paginationWrapper.exists()).toBeTruthy();
    expect(paginationWrapper.text()).toEqual('1 / 3');
  });

  it('should not render pagination', () => {
    const wrapper = mount(<ImagePreview visible images={images} showPagination={false} />);
    expect(wrapper.find('.za-image-preview__index').exists()).toBeFalsy();
  });

  it('should render images', () => {
    const wrapper = mount(<ImagePreview visible images={images} />);
    const itemWrapper = wrapper.find('.za-image-preview__item');
    expect(itemWrapper).toHaveLength(3);
    const srcArr = itemWrapper.map((w) => w.find('img')).map((v) => v.prop('src'));
    expect(srcArr).toEqual(images);
  });
});
