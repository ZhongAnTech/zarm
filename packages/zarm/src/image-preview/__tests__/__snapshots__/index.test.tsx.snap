// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview snapshot renders correctly 1`] = `
<ImagePreview
  activeIndex={0}
  images={
    Array [
      "https://static.zhongan.com/website/health/zarm/images/banners/1.png",
      "https://static.zhongan.com/website/health/zarm/images/banners/2.png",
      "https://static.zhongan.com/website/health/zarm/images/banners/3.png",
    ]
  }
  onChange={[MockFunction]}
  prefixCls="za-image-preview"
  showPagination={true}
  visible={true}
>
  <ForwardRef(ForwardedRefComp)
    className="za-image-preview"
    direction="center"
    visible={true}
  >
    <WarnIfDeprecatedComp
      className="za-image-preview"
      direction="center"
      forwardedRef={null}
      visible={true}
    >
      <Popup
        className="za-image-preview"
        destroy={true}
        direction="center"
        prefixCls="za-popup"
        visible={true}
      >
        <Portal
          animationDuration={200}
          animationType="fade"
          className="za-image-preview"
          destroy={true}
          direction="center"
          handlePortalUnmount={[Function]}
          mask={true}
          maskType="normal"
          prefixCls="za-popup"
          visible={true}
        >
          <Trigger
            disabled={false}
            onClose={[Function]}
            visible={true}
          >
            <Portal
              containerInfo={
                <div
                  class="za-popup-container za-image-preview"
                >
                  <div
                    class="za-mask za-fade-enter za-mask--normal"
                    style="animation-duration: 200ms;"
                  />
                  <div
                    class="za-popup__wrapper za-fade-enter"
                    data-width="0"
                    role="dialog"
                    style="animation-duration: 200ms;"
                  >
                    <div
                      class="za-popup za-popup--center za-fade-enter"
                      data-width="0"
                      role="document"
                      style="animation-duration: 200ms;"
                    >
                      <div
                        class="za-image-preview__content"
                      >
                        <div
                          class="za-carousel za-carousel--horizontal"
                        >
                          <div
                            class="za-carousel__items"
                            style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
                          >
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://static.zhongan.com/website/health/zarm/images/banners/1.png"
                                />
                              </div>
                            </div>
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://static.zhongan.com/website/health/zarm/images/banners/2.png"
                                />
                              </div>
                            </div>
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://static.zhongan.com/website/health/zarm/images/banners/3.png"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="za-image-preview__footer"
                      >
                        <div
                          class="za-image-preview__index"
                        >
                          1
                           / 
                          3
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              }
            >
              <Mask
                className="za-fade-enter"
                prefixCls="za-mask"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
                type="normal"
                visible={true}
              >
                <div
                  className="za-mask za-fade-enter za-mask--normal"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                />
              </Mask>
              <div
                className="za-popup__wrapper za-fade-enter"
                data-width={0}
                onClick={[Function]}
                role="dialog"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
              >
                <div
                  className="za-popup za-popup--center za-fade-enter"
                  data-width={0}
                  role="document"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                      "width": undefined,
                    }
                  }
                >
                  <div
                    className="za-image-preview__content"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                  >
                    <Carousel
                      activeIndex={0}
                      animationDuration={500}
                      autoPlay={false}
                      autoPlayIntervalTime={3000}
                      direction="left"
                      height={160}
                      loop={false}
                      moveDistanceRatio={0.5}
                      moveTimeSpan={300}
                      onChange={[Function]}
                      prefixCls="za-carousel"
                      showPagination={false}
                      swipeable={true}
                    >
                      <div
                        className="za-carousel za-carousel--horizontal"
                      >
                        <Drag
                          onDragEnd={[Function]}
                          onDragMove={[Function]}
                          onDragStart={[Function]}
                        >
                          <div
                            className="za-carousel__items"
                            onMouseDown={[Function]}
                            onMouseMove={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            onTransitionEnd={[Function]}
                            style={Object {}}
                          >
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$0"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://static.zhongan.com/website/health/zarm/images/banners/1.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$1"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://static.zhongan.com/website/health/zarm/images/banners/2.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$2"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://static.zhongan.com/website/health/zarm/images/banners/3.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                          </div>
                        </Drag>
                      </div>
                    </Carousel>
                  </div>
                  <div
                    className="za-image-preview__footer"
                  >
                    <div
                      className="za-image-preview__index"
                    >
                      1
                       / 
                      3
                    </div>
                  </div>
                </div>
              </div>
            </Portal>
          </Trigger>
        </Portal>
      </Popup>
    </WarnIfDeprecatedComp>
  </ForwardRef(ForwardedRefComp)>
</ImagePreview>
`;

exports[`ImagePreview snapshot renders correctly with origin 1`] = `
<ImagePreview
  activeIndex={0}
  images={
    Array [
      Object {
        "originUrl": "https://static.zhongan.com/website/health/zarm/images/banners/1.png",
        "url": "https://cdn-health.zhongan.com/zarm/imagePreview/compress_1.png",
      },
      Object {
        "originUrl": "https://static.zhongan.com/website/health/zarm/images/banners/2.png",
        "url": "https://cdn-health.zhongan.com/zarm/imagePreview/compress_2.png",
      },
      Object {
        "originUrl": "https://static.zhongan.com/website/health/zarm/images/banners/3.png",
        "url": "https://cdn-health.zhongan.com/zarm/imagePreview/compress_3.png",
      },
    ]
  }
  onChange={[MockFunction]}
  prefixCls="za-image-preview"
  showPagination={true}
  visible={true}
>
  <ForwardRef(ForwardedRefComp)
    className="za-image-preview"
    direction="center"
    visible={true}
  >
    <WarnIfDeprecatedComp
      className="za-image-preview"
      direction="center"
      forwardedRef={null}
      visible={true}
    >
      <Popup
        className="za-image-preview"
        destroy={true}
        direction="center"
        prefixCls="za-popup"
        visible={true}
      >
        <Portal
          animationDuration={200}
          animationType="fade"
          className="za-image-preview"
          destroy={true}
          direction="center"
          handlePortalUnmount={[Function]}
          mask={true}
          maskType="normal"
          prefixCls="za-popup"
          visible={true}
        >
          <Trigger
            disabled={false}
            onClose={[Function]}
            visible={true}
          >
            <Portal
              containerInfo={
                <div
                  class="za-popup-container za-image-preview"
                >
                  <div
                    class="za-mask za-fade-enter za-mask--normal"
                    style="animation-duration: 200ms;"
                  />
                  <div
                    class="za-popup__wrapper za-fade-enter"
                    data-width="0"
                    role="dialog"
                    style="animation-duration: 200ms;"
                  >
                    <div
                      class="za-popup za-popup--center za-fade-enter"
                      data-width="0"
                      role="document"
                      style="animation-duration: 200ms;"
                    >
                      <div
                        class="za-image-preview__content"
                      >
                        <div
                          class="za-carousel za-carousel--horizontal"
                        >
                          <div
                            class="za-carousel__items"
                            style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
                          >
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_1.png"
                                />
                              </div>
                            </div>
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_2.png"
                                />
                              </div>
                            </div>
                            <div
                              class="za-carousel__item za-image-preview__item"
                            >
                              <div
                                class="za-image-preview__item__img za-pinch-zoom"
                              >
                                <img
                                  alt=""
                                  draggable="false"
                                  src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_3.png"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="za-image-preview__footer"
                      >
                        <button
                          class="za-image-preview__origin__button"
                        />
                        <div
                          class="za-image-preview__index"
                        >
                          1
                           / 
                          3
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              }
            >
              <Mask
                className="za-fade-enter"
                prefixCls="za-mask"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
                type="normal"
                visible={true}
              >
                <div
                  className="za-mask za-fade-enter za-mask--normal"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                />
              </Mask>
              <div
                className="za-popup__wrapper za-fade-enter"
                data-width={0}
                onClick={[Function]}
                role="dialog"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
              >
                <div
                  className="za-popup za-popup--center za-fade-enter"
                  data-width={0}
                  role="document"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                      "width": undefined,
                    }
                  }
                >
                  <div
                    className="za-image-preview__content"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                  >
                    <Carousel
                      activeIndex={0}
                      animationDuration={500}
                      autoPlay={false}
                      autoPlayIntervalTime={3000}
                      direction="left"
                      height={160}
                      loop={false}
                      moveDistanceRatio={0.5}
                      moveTimeSpan={300}
                      onChange={[Function]}
                      prefixCls="za-carousel"
                      showPagination={false}
                      swipeable={true}
                    >
                      <div
                        className="za-carousel za-carousel--horizontal"
                      >
                        <Drag
                          onDragEnd={[Function]}
                          onDragMove={[Function]}
                          onDragStart={[Function]}
                        >
                          <div
                            className="za-carousel__items"
                            onMouseDown={[Function]}
                            onMouseMove={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            onTransitionEnd={[Function]}
                            style={Object {}}
                          >
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$0"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_1.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$1"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_2.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                            <div
                              className="za-carousel__item za-image-preview__item"
                              key=".$2"
                            >
                              <PinchZoom
                                className="za-image-preview__item__img"
                                maxScale={3}
                                minScale={1}
                                prefixCls="za-pinch-zoom"
                              >
                                <div
                                  className="za-image-preview__item__img za-pinch-zoom"
                                >
                                  <img
                                    alt=""
                                    draggable={false}
                                    key="0/.0"
                                    onLoad={[Function]}
                                    src="https://cdn-health.zhongan.com/zarm/imagePreview/compress_3.png"
                                  />
                                </div>
                              </PinchZoom>
                            </div>
                          </div>
                        </Drag>
                      </div>
                    </Carousel>
                  </div>
                  <div
                    className="za-image-preview__footer"
                  >
                    <button
                      className="za-image-preview__origin__button"
                      onClick={[Function]}
                    />
                    <div
                      className="za-image-preview__index"
                    >
                      1
                       / 
                      3
                    </div>
                  </div>
                </div>
              </div>
            </Portal>
          </Trigger>
        </Portal>
      </Popup>
    </WarnIfDeprecatedComp>
  </ForwardRef(ForwardedRefComp)>
</ImagePreview>
`;
